WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomerSection/OrderCustomerSection.swift

```diff
@@ -0,0 +1,141 @@
+import SwiftUI
+import Networking
+
+/// Represents the Customer section
+///
+struct OrderCustomerSection: View {
+    let geometry: GeometryProxy
+
+    /// View model to drive the view content
+    @ObservedObject var viewModel: NewOrderViewModel
+
+    @State private var showAddressForm: Bool = false
+
+    var body: some View {
+        Group {
+            Divider()
+
+            VStack(alignment: .leading) {
+                HStack(alignment: .top) {
+                    Text(Localization.customer)
+                        .headlineStyle()
+
+                    Spacer()
+
+                    if viewModel.customerDataViewModel.isDataAvailable {
+                        Button(Localization.editButton) {
+                            showAddressForm.toggle()
+                        }
+                        .buttonStyle(LinkButtonStyle())
+                        .fixedSize(horizontal: true, vertical: true)
+                        .padding(.top, -Layout.linkButtonTopPadding) // remove padding to align button title to the top
+                        .padding(.trailing, -Layout.linkButtonTrailingPadding) // remove padding to align button title to the side
+                    }
+                }
+
+                if !viewModel.customerDataViewModel.isDataAvailable {
+                    createCustomerView
+                } else {
+                    if let fullName = viewModel.customerDataViewModel.fullName {
+                        Text(fullName)
+                            .bodyStyle()
+
+                        if let email = viewModel.customerDataViewModel.email {
+                            Spacer()
+                            Text(email)
+                                .footnoteStyle()
+                        }
+                    }
+                    if let billingAddressFormatted = viewModel.customerDataViewModel.billingAddressFormatted {
+                        Divider()
+                        Text(Localization.billingTitle)
+                            .headlineStyle()
+                        Spacer()
+                        Text(billingAddressFormatted)
+                            .bodyStyle()
+                    }
+                    if let shippingAddressFormatted = viewModel.customerDataViewModel.shippingAddressFormatted {
+                        Divider()
+                        Text(Localization.shippingTitle)
+                            .headlineStyle()
+                        Spacer()
+                        Text(shippingAddressFormatted)
+                            .bodyStyle()
+                    }
+                }
```

Comment:
Nit: This seems like it could have been extracted to another view just to make it more readable.

Reply:
Applied some refactoring in 332e9f8d7abad7f928fdc2a173e89725414b1a10.